hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -470,6 +481,22 @@ private void setHmsTableParameters(
     // remove any props from HMS that are no longer present in Iceberg table props
     obsoleteProps.forEach(parameters::remove);
 
+    if (metadata.property(HiveCatalog.HMS_TABLE_OWNER, null) != null) {
```

Comment:
Looks like this code is for the update case and is going a bit beyond original behavior (which was for create case).  

cc @gaborkaszab for thoughts

And does it align with database/namespace update?

Reply:
Agree @szehon-ho that this seems a different use-case for me. Might worth covering it in a separate PR.

Additionally, this function is for setting the table parameters and with this change it would also have a side effect of setting the tbl's owner that is not really a table property. I would feel this is a bit off in my opinion.

And anyway, when this code was run for a "create table" scenario then newHmsTable() would have already set the table owner so the below line would be redundant.

But first of all as this is for the update scenario I'd say that this deserve a separate PR because this one is for changing the defaults during table creation.

Reply:
cool, I can separate it out into a separate PR. I hope to discuss whether this feature is needed. Basically, what I'm trying to achieve here is users can tweak ownership by setting iceberg table properties. And should users remove the owner properties, then it falls back to default owner.

Reply:
I put it there cause I was thinking to unify the ownership behavior between database and table.